_episodes/07-thresholding.md

```diff
@@ -14,532 +14,370 @@ threshold an image represented by a numpy array."
-value for `t`. How might we do that? Well, one way is to look at a grayscale
-histogram of the image. Here is the histogram produced by the 
-**GrayscaleHistogram.py** program from the 
-[Creating Histograms]({{ page.root }}/05-creating-histograms/) episode, if we
-run it on the colored shapes image shown above.
-
-![Grayscale histogram](../fig/06-junk-histogram.png)
-
-Since the image has a white background, most of the pixels in the image are 
-white. This corresponds nicely to what we see in the histogram: there is a
-spike near the value of 1.0. If we want to select the shapes and not the
-background, we want to turn off the white background pixels, while leaving the
-pixels for the shapes turned on. So, we should choose a value of `t` somewhere
-before the large peak and turn pixels above that value "off".
-
-Here are the first few lines of a Python program to apply simple thresholding to the image, to 
-accomplish this task. 
-
-~~~
-"""
- * Python script to demonstrate simple thresholding.
- *
- * usage: python Threshold.py <filename> <sigma> <threshold>
- * Example parameter values: 2 for sigma and .8 for threshold
-"""
-import sys
-import numpy as np
-import skimage.color
-import skimage.filters
-import skimage.io
+it to grayscale, and de-noise it as in the [Blurring]({{ page.root }}/06-blurring/) episode.
+
+~~~
+# convert the image to grayscale
+gray_image = skimage.color.rgb2gray(image)
 
-# get filename, sigma, and threshold value from command line
-filename = sys.argv[1]
-my_sigma = float(sys.argv[2])
-t = float(sys.argv[3])
+# blur the image to denoise
+blurred_image = skimage.filters.gaussian(gray_image, sigma=1.0)
 
-# read and display the original image
-image = skimage.io.imread(fname=filename)
-skimage.io.imshow(image)
+fig, ax = plt.subplots()
+plt.imshow(blurred_image, cmap='gray')
+plt.show()
 ~~~
 {: .language-python}
 
-This program takes three command-line arguments: the filename of the image to 
-manipulate, the sigma of the Gaussian used during the blurring step (which, if you recall
-from the [Blurring]({{ page.root }}/06-blurring/) episode, must be a float),
-and finally, the threshold value `t`, which should be a float in the closed
-range [0.0, 1.0]. The program takes the command-line values and stores them in
-variables named `filename`, `my_sigma`, and `t`, respectively.
+![Grayscale image of the geometric shapes](../../fig/06-junk-grayscale.png)
 
-Next, the program reads the original image based on the `filename` value, and
-displays it. 
+Next, we would like to apply the threshold `t`, a number in the closed range [0.0, 1.0]. Pixels with color values on one  side of `t` will be turned "on," while pixels with color values on the other side will be turned "off." To use this process, we first have to determine a "good" value for `t`. How might we do that? One way is to look at the grayscale histogram of the image and try to identify what grayscale ranges correspond to the shapes in the image or the background.
```

Comment:
> a number in the closed range [0.0, 1.0]

I think this would benefit from an explanation that `t` must be within the range of the values included in the image array - at the moment I feel like this is suggesting that a threshold must always be between 0.0. and 1.0, which will become confusing when trying to apply the same principle to an image with 0-255 values.

Reply:
Thanks for the suggestion @tobyhodges. I agree that some more explanation would be good. This was also suggested in #32 which is still on my todo list.

Reply:
@ErinBecker Hi Erin, thanks for bringing up the question what makes a good threshold. I have added a callout in c51636c to discuss this a bit more. Do you think this resolves #32?